### PR TITLE
Moves sorting of crew console listings to clientside + refactors

### PR DESCRIPTION
**Pull request by bobbah**
## About The Pull Request

Moves the sorting of the entries within the crew console to be clientside. There's no reason to have this serverside.

I've also refactored the UI data code for the crew console in general, reduced code duplication and reduced unnecessary data transfer of nulls.

## Why It's Good For The Game

Improved maintainability, hopefully better performance. Mentioned by @stylemistake.

## Changelog
:cl: bobbahbrown
refactor: Cleaned up the crew console code, moved sorting of sensor entries to clientside.
/:cl:

### CODE REVIEW

**Review by stylemistake — DISMISSED** [review]

File: code/game/machinery/computer/crew.dm [hunk 1]
```diff
@@ -97,128 +96,90 @@ GLOBAL_DATUM_INIT(crewmonitor, /datum/crewmonitor, new)
...
-					pos_x = null
-					pos_y = null
-
-				results[++results.len] = list("name" = name, "assignment" = assignment, "ijob" = ijob, "life_status" = life_status, "oxydam" = oxydam, "toxdam" = toxdam, "burndam" = burndam, "brutedam" = brutedam, "area" = area, "pos_x" = pos_x, "pos_y" = pos_y, "can_track" = H.can_track(null))
-
-	data_by_z["[z]"] = sortTim(results,/proc/sensor_compare)
+		// Check if z-level is correct
+		// Accept z == 0 as well in case mob is inside an object
+		if (H.z != 0 && H.z != z)
+			continue
```
Comment (Jared-Fogle) [inline code/game/machinery/computer/crew.dm:115]: Why not have this and everything else that checks `z != 0` just use `get_turf(H).z`?

File: code/game/machinery/computer/crew.dm [hunk 2]
```diff
@@ -97,128 +96,90 @@ GLOBAL_DATUM_INIT(crewmonitor, /datum/crewmonitor, new)
...
-	var/burndam
-	var/brutedam
-	var/area
-	var/pos_x
-	var/pos_y
-	var/life_status
-
-	for(var/i in GLOB.nanite_sensors_list)
+	for(var/i in GLOB.suit_sensors_list | GLOB.nanite_sensors_list)
 		var/mob/living/carbon/human/H = i
```
Comment (Jared-Fogle) [inline code/game/machinery/computer/crew.dm:110]: Better variable names.

File: code/game/machinery/computer/crew.dm [hunk 3]
```diff
@@ -97,128 +96,90 @@ GLOBAL_DATUM_INIT(crewmonitor, /datum/crewmonitor, new)
...
+
+		// The entry for this human
+		var/list/entry = list(
+			"ref" = REF(H)
+		)
+
+		// ID and id-related data
+		var/name = "Unknown"
+		var/assignment
+		var/ijob = 81
```
Comment (Jared-Fogle) [inline code/game/machinery/computer/crew.dm:143]: Magic number, and an especially non obvious one.

File: code/game/machinery/computer/crew.dm [hunk 4]
```diff
@@ -97,128 +96,90 @@ GLOBAL_DATUM_INIT(crewmonitor, /datum/crewmonitor, new)
...
+		var/obj/item/card/id/id_card = H.wear_id ? H.wear_id.GetID() : null
```
Comment (Jared-Fogle) [inline code/game/machinery/computer/crew.dm:144]:
```suggestion
		var/obj/item/card/id/id_card = H.wear_id?.GetID()
```

Comment (Jared-Fogle) [reply code/game/machinery/computer/crew.dm:144]: This allows it to work with IDs on peoples belts first. If this is not the behavior you want, `H.wear_id?.GetID()` is a cleaner form of your current code.

File: code/game/machinery/computer/crew.dm [hunk 5]
```diff
@@ -97,128 +96,90 @@ GLOBAL_DATUM_INIT(crewmonitor, /datum/crewmonitor, new)
...
+			entry += list(
+				"oxydam" = round(H.getOxyLoss(), 1),
+				"toxdam" = round(H.getToxLoss(), 1),
+				"burndam" = round(H.getFireLoss(), 1),
+				"brutedam" = round(H.getBruteLoss(), 1)
+			)
+
+		// Location
+		if (pos && (using_nanites || uniform.sensor_mode >= SENSOR_COORDS))
+			entry["area"] = get_area_name(H, TRUE)
```
Comment (Jared-Fogle) [inline code/game/machinery/computer/crew.dm:170]:
```suggestion
			entry["area"] = get_area_name(H, format_text = TRUE)
```

Personal qualm--non-obvious parameters should be named for easy readability.

File: code/game/machinery/computer/crew.dm [hunk 6]
```diff
@@ -97,128 +96,90 @@ GLOBAL_DATUM_INIT(crewmonitor, /datum/crewmonitor, new)
...
+
+		// Trackability
+		entry["can_track"] = H.can_track(null)
```
Comment (Jared-Fogle) [inline code/game/machinery/computer/crew.dm:173]:
```suggestion
		entry["can_track"] = H.can_track()
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Mostly just remnants of the older code.

**Comment by bobbahbrown** [comment]
resolved in latest

**Pull request merged**